coordinator/src/collaborative_revert.rs

```diff
@@ -25,257 +26,280 @@ use dlc_manager::subchannel::LnDlcChannelSigner;
+/// (the funding TXO) and 2 outputs, one for each party.
+///
+/// If either party were to _not_ have an output, we would be overestimating the weight of the
+/// transaction and would end up paying higher fees than necessary.
 const COLLABORATIVE_REVERT_TX_WEIGHT: usize = 672;
 
-pub async fn notify_user_to_collaboratively_revert(
-    revert_params: Json<CollaborativeRevert>,
-    channel_id_string: String,
-    channel_id: [u8; 32],
-    pool: Pool<ConnectionManager<PgConnection>>,
+/// Send a collaborative revert proposal to the counterparty in the channel identified by
+/// `channel_id`.
+///
+/// A collaborative revert involves signing a new transaction spending from the funding output
+/// directly. This can be used to circumvent bugs related to position and subchannel state.
+///
+/// The `without_channel_details_params` argument is optional because one should rely on the values
+/// provided by the [`ChannelManager`] _if possible_. If LDK has already attempted to close the
+/// channel, it will be impossible to fetch its [`ChannelDetails`] and the caller will have to
+/// provide this data.
+#[allow(clippy::too_many_arguments)]
+pub async fn propose_collaborative_revert(
     node: Arc<Node<CoordinatorTenTenOneStorage, NodeStorage>>,
-    auth_users_notifier: mpsc::Sender<OrderbookMessage>,
-) -> anyhow::Result<()> {
-    let mut conn = pool.get().context("Could not acquire db lock")?;
+    pool: Pool<ConnectionManager<PgConnection>>,
+    sender: mpsc::Sender<OrderbookMessage>,
+    channel_id: [u8; 32],
+    settlement_price: Decimal,
+    fee_rate_sats_vb: u64,
+    funding_txo: OutPoint,
+    without_channel_details_params: Option<WithoutChannelDetailsParams>,
+) -> Result<()> {
+    let mut conn = pool.get().context("Could not acquire DB lock")?;
 
-    let channel_details = node
-        .channel_manager
-        .get_channel_details(&channel_id)
-        .context("Could not get channel")?;
+    let channel_id_hex = channel_id.to_hex();
+
+    let channel_details_option = node.channel_manager.get_channel_details(&channel_id);
+    let (ln_inbound_liquidity_sat, ln_outbound_liquidity_sat) = match channel_details_option {
+        Some(ref details) => {
+            let ln_inbound_liquidity_sat =
+                Decimal::from(details.inbound_capacity_msat) / Decimal::ONE_THOUSAND;
+            let ln_inbound_liquidity_sat = ln_inbound_liquidity_sat
+                .to_u64()
+                .expect("inbound liquidity to fit into u64");
+
+            let ln_outbound_liquidity_sat =
+                Decimal::from(details.outbound_capacity_msat) / Decimal::ONE_THOUSAND;
+            let ln_outbound_liquidity_sat = ln_outbound_liquidity_sat
+                .to_u64()
+                .expect("outbound liquidity to fit into u64");
+
+            (ln_inbound_liquidity_sat, ln_outbound_liquidity_sat)
+        }
+        None => {
+            let params = without_channel_details_params.context(
```

Comment:
🤔 would it make sense to use the inbound and outbound capacity from the shadow channel? I feel like this would anyways the spot where I'd be looking for these values.

Reply:
It's a fair point. I sort of assumed that the shadow channel would be unreliable, but we seem to do our best to keep it synchronised.